From de5e1f920208888ca7941d7855689ba6228387bf Mon Sep 17 00:00:00 2001
From: Shaun Tancheff <shaun@tancheff.com>
Date: Mon, 14 Nov 2016 22:22:19 -0600
Subject: [PATCH 11/11] Seagate Internal: Patches for debian unstable gcc-6

Debian unstable changed the default build of gcc-6 which breaks the
default kernel build.

Added -fno-pic / -fno-pie to working gcc-6 config defaults.

Signed-off-by: Shaun Tancheff <shaun.tancheff@seagate.com>
---
 Makefile                                  | 5 +++++
 scripts/gcc-x86_64-has-stack-protector.sh | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 80b8671..260ae76 100644
--- a/Makefile
+++ b/Makefile
@@ -619,6 +619,11 @@ ARCH_AFLAGS :=
 ARCH_CFLAGS :=
 include arch/$(SRCARCH)/Makefile
 
+KBUILD_CPPFLAGS += $(call cc-option,-fno-pic,)
+KBUILD_CPPFLAGS += $(call cc-option,-fno-pie,)
+
+KBUILD_CFLAGS	+= $(call cc-option,-fno-pic,)
+KBUILD_CFLAGS	+= $(call cc-option,-fno-pie,)
 KBUILD_CFLAGS	+= $(call cc-option,-fno-delete-null-pointer-checks,)
 KBUILD_CFLAGS	+= $(call cc-disable-warning,maybe-uninitialized,)
 
diff --git a/scripts/gcc-x86_64-has-stack-protector.sh b/scripts/gcc-x86_64-has-stack-protector.sh
index 973e8c1..274cba1 100755
--- a/scripts/gcc-x86_64-has-stack-protector.sh
+++ b/scripts/gcc-x86_64-has-stack-protector.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-echo "int foo(void) { char X[200]; return 3; }" | $* -S -x c -c -O0 -mcmodel=kernel -fstack-protector - -o - 2> /dev/null | grep -q "%gs"
+echo "int foo(void) { char X[200]; return 3; }" | $* -S -x c -c -O0 -mcmodel=kernel -fno-pie -fstack-protector - -o - 2> /dev/null | grep -q "%gs"
 if [ "$?" -eq "0" ] ; then
 	echo y
 else
-- 
2.10.2

